src/pages/ProfilesExplorerView/exploration-types/SceneServiceFlameGraph/SceneFlameGraph.tsx

```diff
@@ -33,6 +35,27 @@ export class SceneFlameGraph extends SceneObjectBase<SceneFlameGraphState> {
         queries: [],
       }),
     });
+
+    this.addActivationHandler(this.onActivate.bind(this));
+  }
+
+  onActivate() {
+    let timeRangeSubscription: Unsubscribable | null = null;
+
+    const stateSubscription = this.subscribeToState((newState) => {
```

Comment:
There are a couple of issues with this implementation:
- `$data` always exists (it's set when calling `super` L33)
- every time the state of this scene object will change, a new subscription will be created (L47) - we only need a single one
- there is still a conceptual problem: do we know under which circumstances `data.timeRange` changes? is it when starting to load the data? after loading? what we really need is to store the last timerange **that was used to successfully load the data**

My recommendation would be to use this implementation:

```js
    let dataSub: undefined | Unsubscribable;

    const stateSub = this.subscribeToState((newState, prevState) => {
      if (newState.$data !== prevState.$data) {
        if (dataSub) {
          dataSub.unsubscribe();
        }

        dataSub = newState.$data?.subscribeToState((newState) => {
          if (newState.data?.state === LoadingState.Done) {
            timelineAndProfileApiClient.setLastTimeRange(newState.data.timeRange);
          }
        });
      }
    });

    return () => {
      stateSub.unsubscribe();
      dataSub?.unsubscribe();
    };
```

Reply:
Ah gotcha, this makes sense. Fixed in https://github.com/grafana/explore-profiles/pull/25/commits/46cfccbb443190e026ceec9a16e4e9f19c4a5990